src/backend/joanie/core/models/wishlist.py

```diff
@@ -0,0 +1,96 @@
+"""
+Declare and configure the models for the wishlist part
+"""
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from . import Course, CourseRun, User
+from .base import BaseModel
+
+
+class Wishlist(BaseModel):
+    """
+    Wishlist represents and records a user wish to participate at a course
+    """
+
+    owner = models.ForeignKey(
+        to=User,
+        verbose_name=_("Owner"),
+        related_name="wishlists",
+        on_delete=models.PROTECT,
+    )
+
+    course = models.ForeignKey(
+        to=Course,
+        verbose_name=_("Course"),
+        related_name="wished_in_wishlists",
+        on_delete=models.PROTECT,
+    )
+
+    course_run_statuses = models.ManyToManyField(
```

Comment:
I think there will be some cases when we need to notify the user not because a new course run was added, but because for example a new product is added to sell it? Or the limit on the product selling this course is raised?
Maybe this object is in fact the Notification object that we were talking about and should be more generic with subject/action/object or something like this? Not sure what is a good patterrn for notification.

Reply:
ok your're right, I will change that !

Reply:
Finally, I remove the "Notification" model from this PR, to take time to think about it. 
Then this PR which contains only the CourseWish model and the corresponding routes can be merged more easily.